js/components/graph/NodeGroup.js

```diff
@@ -48,80 +48,28 @@ export default class NodeGroup extends React.Component {
-          this.props.edgesJSON.map(element => {
-            // Note: hybrids shouldn't have any in edges
-            if (entry.id_ === element.source) {
-              childs.push(element.target);
-              outEdges.push(element.id_);
-            }
-          });
-          // parse prereqs based on text
-          var hybridText = "";
-          entry.text.forEach(textTag => (hybridText += textTag.text));
-          var parents = [];
-          // First search for entire string (see Stats graph)
-          var prereqNode = this.findRelationship(hybridText);
-          if (prereqNode !== undefined) {
-            parents.push(prereqNode.id_);
-            hybridRelationships.push([prereqNode.id_, entry.id_]);
-          } else {
-            // Parse text first
-            var prereqs = parseAnd(hybridText)[0];
-            prereqs.forEach(course => {
-              if (typeof course === "string") {
-                prereqNode = this.findRelationship(course);
-                if (prereqNode !== undefined) {
-                  parents.push(prereqNode.id_);
-                  hybridRelationships.push([prereqNode.id_, entry.id_]);
-                } else {
-                  console.error("Could not find prereq for ", hybridText);
-                }
-              } else if (typeof course === "object") {
-                var orPrereq = [];
-                course.forEach(c => {
-                  var prereqNode = this.findRelationship(c);
-                  if (prereqNode !== undefined) {
-                    orPrereq.push(prereqNode.id_);
-                    hybridRelationships.push([prereqNode.id_, entry.id_]);
-                  } else {
-                    console.error("Could not find prereq for ", hybridText);
-                  }
-                });
-                if (orPrereq.length > 0) {
-                  parents.push(orPrereq);
-                }
-              }
-            });
-          }
           return (
             <Node
               JSON={entry}
               className={"hybrid"}
               key={entry.id_}
               hybrid={true}
               ref={this.setRefEntry(entry)}
-              parents={parents}
-              childs={childs}
+              parents={this.props.connections.parents[entry.id_]}
+              childs={this.props.connections.children[entry.id_]}
               inEdges={[]}
-              outEdges={outEdges}
+              outEdges={this.props.connections.outEdges[entry.id_]}
               svg={svg}
               logicalType={"AND"}
             />
           );
         })}
         {this.props.nodesJSON.map(entry => {
           var highlighted = highlightedNodes.indexOf(entry.id_) >= 0;
-          hybridRelationships.forEach(element => {
-            if (element[0] === entry.id_) {
-              this.props.connections.children[entry.id_].push(element[1]);
-            }
-          });
+          var childs = this.props.connections.children[entry.id_].slice();
```

Comment:
This is interesting, I'm not sure why creating a copy (using `slice`) is necessary. Is this list being mutated somewhere when rendering the `Node`?

Reply:
This is a remnant from before the changes this PR. The copying was to stop the issue where nodes that correspond to hybrids would get their hybrid children re-added every time `render()` ran. I'm removing it now that we no longer have `hybridRelationships`.